src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -711,6 +711,17 @@ private void GenerateClassInternalsField(Field field)
                 !(@class != null && @class.IsUnion)) || (@class != null && @class.TranslationUnit.IsSystemHeader))
                 return;
 
+            if (field.Class.Fields.Count(c => c.Name.Equals(field.Name)) > 1)
```

Comment:
We should not modify `field.OriginalName`. So I'd advise to extract this piece of code to a static method and change the `Helpers.SafeIdentifier(field.OriginalName);` call to call the method instead.